src/main/java/gift/ProductAdminController.java

```diff
@@ -0,0 +1,86 @@
+package gift;
+
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@Controller
+@RequestMapping("/api/admin")
+public class ProductAdminController {
+    private final ProductDao productdao;
+
+    public ProductAdminController(ProductDao productdao) {
+        this.productdao = productdao;
+    }
+
+    @GetMapping("/product/list")
+    public String findAll(Model model) {
+        List<Product> products = productdao.getAll();
+        model.addAttribute("products", products);
+        return "products";
+    }
+
+    @GetMapping("/product/add")
+    public String addForm(Model model) {
+        model.addAttribute("productdto", new ProductDto());
+        return "addForm";
+    }
+
+    @PostMapping("/product/add")
+    public String saveProduct(@ModelAttribute ProductDto productdto) {
+        ProductDto productdto1;
+        productdto1 = new ProductDto(productdto.getName(), productdto.getPrice(), productdto.getImageUrl());
+        productdao.save(productdto1);
+        return "redirect:/api/admin/product/list";
+    }
+
+    @ResponseBody
+    @GetMapping("/product/{id}")
+    public Product findById(@PathVariable Long id) {
+        return productdao.findById(id);
+    }
+
+    @GetMapping("/product/{id}/update")
+    public String updateForm(@PathVariable Long id, Model model) {
+        Product product = productdao.findById(id);
+        model.addAttribute("product", product);
+        return "updateForm";
+    }
+
+    @PatchMapping("/product/{id}/update")
+    public String updateProduct(@PathVariable Long id, @ModelAttribute ProductDto updateProductdto) {
+        ProductDto productdto1;
+        productdto1 = new ProductDto(updateProductdto.getName(), updateProductdto.getPrice(), updateProductdto.getImageUrl());
```

Comment:
```suggestion
        ProductDto productdto1 = new ProductDto(updateProductdto.getName(), updateProductdto.getPrice(), updateProductdto.getImageUrl());
```
inline 해서 사용해도 좋을 것 같아요~ 또는 ProductDto를 인자로 받는 생성자를 만들어도 좋구요. 근데 사용성이 조금 이상할 수 있겠네요 😢 질문드린 내용에 대한 답변이랑 같이 메인 comment에 남길게요!